### ISSUE TITLE
ofi: provider verbs;ofi_rxm with mlx hangs on osu_acc_lantency

### ISSUE THREAD

**Issue by aingerson** [pmodels/mpich#4187]
I'm getting the following hang on the osu_acc_latency only when running verbs;ofi_rxm over mlx not over hfi. Is this known/is there something extra I need to set?

mpirun -n 2 -ppn 1 osu_acc_latency
OSU MPI_Accumulate latency Test v5.5
Window creation: MPI_Win_allocate
Synchronization: MPI_Win_flush
Size          Latency (us)
1                       2.95
2                       2.93
4                       2.93
8                       2.92
16                      2.94
32                      2.98
64                      3.05
128                     3.42
256                     3.50
512                     3.73
1024                    4.22
2048                    5.10
4096                    7.00
8192                   10.49
16384                  16.25
32768                  26.32
65536                  45.70
131072                 88.13
262144                161.30
<hang>

**Comment by wesbland** [comment]
It's possible that this is related to #4188

**Comment by aingerson** [comment]
This is fixed by #4197 , specifcally commit 2980cf1e71a0eb006787dd7ff93c6224f22a24fa
Based on the commit message, I'm guessing it was somehow selecting an incorrect provider. Not sure exactly why, but applying that commit gets rid of the hang on MLX.

**Comment by hzhou** [comment]
Thanks @aingerson for confirmation. I'll add this issue to the fixes list in #4197.

Could you also test it while setting `MPIR_CVAR_CH4_OFI_CAPABILITY_SETS_DEBUG=1` and pasting the printout here? Since it fails the default capability set matching (assume that resulted in the bug manifest) I suspect it there may be some issues in the MPICH matching logic, similar to issue #4200.

**Comment by aingerson** [comment]
Before fix/With hang:
```
==== Capability set configuration ====
MPIDI_OFI_ENABLE_AV_TABLE: 0
MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS: 0
MPIDI_OFI_ENABLE_SHARED_CONTEXTS: 0
MPIDI_OFI_ENABLE_MR_SCALABLE: 0
MPIDI_OFI_ENABLE_TAGGED: 1
MPIDI_OFI_ENABLE_AM: 1
MPIDI_OFI_ENABLE_RMA: 1
MPIDI_OFI_ENABLE_ATOMICS: 0
MPIDI_OFI_FETCH_ATOMIC_IOVECS: 1
MPIDI_OFI_IOVEC_ALIGN: 1
MPIDI_OFI_ENABLE_DATA_AUTO_PROGRESS: 0
MPIDI_OFI_ENABLE_CONTROL_AUTO_PROGRESS: 0
MPIDI_OFI_ENABLE_PT2PT_NOPACK: 1
MPIDI_OFI_NUM_AM_BUFFERS: 8
MPIDI_OFI_CONTEXT_BITS: 20
MPIDI_OFI_SOURCE_BITS: 0
MPIDI_OFI_TAG_BITS: 31
======================================
MAXIMUM SUPPORTED RANKS: 4294967296
MAXIMUM TAG: 2147483648
```

After fix/No hang:
```
==== Capability set configuration ====
MPIDI_OFI_ENABLE_AV_TABLE: 1
MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS: 0
MPIDI_OFI_ENABLE_SHARED_CONTEXTS: 0
MPIDI_OFI_ENABLE_MR_SCALABLE: 0
MPIDI_OFI_ENABLE_TAGGED: 0
MPIDI_OFI_ENABLE_AM: 1
MPIDI_OFI_ENABLE_RMA: 0
MPIDI_OFI_ENABLE_ATOMICS: 0
MPIDI_OFI_FETCH_ATOMIC_IOVECS: 1
MPIDI_OFI_IOVEC_ALIGN: 1
MPIDI_OFI_ENABLE_DATA_AUTO_PROGRESS: 0
MPIDI_OFI_ENABLE_CONTROL_AUTO_PROGRESS: 0
MPIDI_OFI_ENABLE_PT2PT_NOPACK: 1
MPIDI_OFI_NUM_AM_BUFFERS: 8
MPIDI_OFI_CONTEXT_BITS: 16
MPIDI_OFI_SOURCE_BITS: 24
MPIDI_OFI_TAG_BITS: 20
======================================
MAXIMUM SUPPORTED RANKS: 16777216
MAXIMUM TAG: 1048576
```

**Comment by aingerson** [comment]
That's with only the one patch, by the way. Not with all 5 patches from your PR. Would you like the output with all 5 applied, too?

**Comment by aingerson** [comment]
With all 5 patches, output is the same except it says before the capabilities:
```
libfabric provider: verbs;ofi_rxm
```

**Comment by hzhou** [comment]
Not really. The rest of the 4 patches are mostly refactoring.

It is using the minimal capability set. I am sure `verbs;ofi_rxm` supports more than that, right? Could you try both
```
MPIR_CVAR_CH4_OFI_CAPABILITY_SETS_DEBUG=1
MPIR_CVAR_OFI_USE_PROVIDER='verbs;ofi_rxm'
```

**Comment by aingerson** [comment]
`MPIR_CVAR_OFI_USE_PROVIDER='verbs;ofi_rxm'` gives the same set of capabilities.

Yes, verbs;ofi_rxm should definitely support tagged messages and RMA

**Comment by hzhou** [comment]
Thanks, I'll attach to the issue #4200

**Comment by hzhou** [comment]
Closed by #4197
